src/duckling/game/drawable/container-drawable.component.ts

```diff
@@ -0,0 +1,139 @@
+import {
+    Component,
+    Input,
+    Output,
+    EventEmitter,
+    forwardRef
+} from '@angular/core';
+import {MD_CARD_DIRECTIVES} from '@angular2-material/card';
+import {MD_LIST_DIRECTIVES} from '@angular2-material/list';
+
+import {CardListElementComponent, FormLabel, EnumChoiceComponent} from '../../controls';
+import {immutableAssign} from '../../util';
+
+
+import {ContainerDrawable} from './container-drawable';
+import {getDefaultDrawable, DrawableComponent} from './drawable.component';
+import {Drawable, DrawableType} from './drawable';
+
+/**
+ * Component used to edit a Container Drawable including all its children drawables
+ */
+@Component({
+    selector: "dk-container-drawable-component",
+    styleUrls: ['./duckling/game/drawable/container-drawable.component.css'],
+    directives: [
+        MD_CARD_DIRECTIVES,
+        CardListElementComponent,
+        FormLabel,
+        EnumChoiceComponent,
+        forwardRef(() => DrawableComponent)],
+    template: `
+        <dk-form-label title="Add Drawable"></dk-form-label>
+        <dk-enum-choice
+            [enum]="DrawableType"
+            [selected]="DrawableType.Shape"
+            (addClicked)="onNewDrawableClicked($event)">
```

Comment:
I would like to refactor this list of card-list-elements out into its own component, but then I don't think I could forward along the contents of each card-list-element (the dk-drawable-component). Do you have any ideas?
 Add a line note